### ISSUE TITLE
textDocument/definition returns wrong path on windows

### ISSUE THREAD

**Issue by prabirshrestha** [palantir/python-language-server#85]
Notice the wrong path where d drive occurs twice. ```{"uri":"file:///D:\\D:\\tmp\\python-helloworld\\helloworld.py"...```. `textDocument/documentSymbol` seems to work.

The log is from `vim-lsp`.

```vim
8/5/2017 3:59:30 PM:["lsp#register_server","server registered","pyls"]
8/5/2017 3:59:30 PM:["lsp#register_server","server registered","rls"]
8/5/2017 3:59:30 PM:["s:on_text_document_did_open()",1]
8/5/2017 3:59:30 PM:[{"response":{"data":{"__data__":"vim-lsp","lps_id":1,"server_name":"pyls"},"message":"started lsp server successfully"}}]
8/5/2017 3:59:30 PM:["--->",1,"pyls",{"method":"initialize","params":{"rootUri":"file:///D:/tmp/python-helloworld","capabilities":{},"rootPath":"file:///D:/tmp/python-helloworld"}}]
8/5/2017 3:59:34 PM:["<---",1,"pyls",{"response":{"id":1,"jsonrpc":"2.0","result":{"capabilities":{"codeLensProvider":{"resolveProvider":false},"executeCommandProvider":{"commands":[]},"hoverProvider":true,"documentFormattingProvider":true,"signatureHelpProvider":{"triggerCharacters":["(",","]},"documentSymbolProvider":true,"definitionProvider":true,"completionProvider":{"resolveProvider":false,"triggerCharacters":["."]},"codeActionProvider":true,"referencesProvider":true,"documentRangeFormattingProvider":true,"textDocumentSync":2}}},"request":{"method":"initialize","jsonrpc":"2.0","id":1,"params":{"rootUri":"file:///D:/tmp/python-helloworld","capabilities":{},"rootPath":"file:///D:/tmp/python-helloworld"}}}]
8/5/2017 3:59:34 PM:["--->",1,"pyls",{"method":"textDocument/didOpen","params":{"textDocument":{"uri":"file:///D:/tmp/python-helloworld/helloworld.py","version":1,"languageId":"python","text":"#!/usr/bin/env python\nclass MyClass:\n    variable = \"blah\"\n\n    def function(self):\n        print(\"This is a message inside the class.\")\n\nmyobjectx = MyClass()\n\nmyobjectx.variable"}}}]
8/5/2017 3:59:34 PM:[{"response":{"data":{"path":"file:///D:/tmp/python-helloworld/helloworld.py","__data__":"vim-lsp","filetype":"python","server_name":"pyls"},"message":"textDocument/open sent"}}]
8/5/2017 3:59:34 PM:[{"response":{"data":{"path":"file:///D:/tmp/python-helloworld/helloworld.py","__data__":"vim-lsp","server_name":"pyls"},"message":"not dirty"}}]
8/5/2017 3:59:34 PM:["<---(stderr)",1,"pyls",["2017-08-05 15:59:34,223 UTC - ERROR - pyls.language_server - CAUGHT\r","Traceback (most recent call last):\r","  File \"c:\\python27\\lib\\site-packages\\pyls\\language_server.py\", line 65, in wrapped\r","    return func(*args, **kwargs)\r","  File \"c:\\python27\\lib\\site-packages\\pyls\\python_ls.py\", line 111, in m_text_document__did_open\r","    self.lint(textDocument['uri'])\r","  File \"c:\\python27\\lib\\site-packages\\pyls\\python_ls.py\", line 93, in lint\r","    self._hooks.pyls_lint, doc_uri\r","  File \"c:\\python27\\lib\\site-packages\\pyls\\python_ls.py\", line 58, in _hook\r","    return hook(config=self.config, workspace=self.workspace, document=doc, **kwargs)\r","  File \"c:\\python27\\lib\\site-packages\\pluggy.py\", line 745, in __call__\r","    return self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)\r","  File \"c:\\python27\\lib\\site-packages\\pluggy.py\", line 339, in _hookexec\r","    return self._inner_hookexec(hook, methods, kwargs)\r","  File \"c:\\python27\\lib\\site-packages\\pluggy.py\", line 302, in __call__\r","    return outcome.get_result()\r","  File \"c:\\python27\\lib\\site-packages\\pluggy.py\", line 280, in get_result\r","    _reraise(*ex)  # noqa\r","  File \"c:\\python27\\lib\\site-packages\\pluggy.py\", line 265, in __init__\r","    self.result = func()\r","  File \"c:\\python27\\lib\\site-packages\\pluggy.py\", line 300, in <lambda>\r","    outcome = _CallOutcome(lambda: self.oldcall(hook, hook_impls, kwargs))\r","  File \"c:\\python27\\lib\\site-packages\\pluggy.py\", line 334, in <lambda>\r","    _MultiCall(methods, kwargs, hook.spec_opts).execute()\r","  File \"c:\\python27\\lib\\site-packages\\pluggy.py\", line 614, in execute\r","    res = hook_impl.function(*args)\r","  File \"c:\\python27\\lib\\site-packages\\pyls\\plugins\\pycodestyle_lint.py\", line 15, in pyls_lint\r","    config_files = config.find_parents(document.path, CONFIG_FILES)\r","  File \"c:\\python27\\lib\\site-packages\\pyls\\config.py\", line 37, in find_parents\r","    return find_parents(self.root_path, path, names)\r","  File \"c:\\python27\\lib\\site-packages\\pyls\\config.py\", line 70, in find_parents\r","    raise ValueError(\"Path %s not in %s\" % (path, root))\r","ValueError: Path /D:/tmp/python-helloworld/helloworld.py not in file:///D:/tmp/python-helloworld\r","2017-08-05 15:59:34,226 UTC - ERROR - jsonrpc.manager - API Exception: {'message': 'Path /D:/tmp/python-helloworld/helloworld.py not in file:///D:/tmp/python-helloworld', 'args': (u'Path /D:/tmp/python-helloworld/helloworld.py not in file:///D:/tmp/python-helloworld',), 'type': 'ValueError'}\r","Traceback (most recent call last):\r","  File \"c:\\python27\\lib\\site-packages\\jsonrpc\\manager.py\", line 108, in _get_responses\r","    result = method(*request.args, **request.kwargs)\r","  File \"c:\\python27\\lib\\site-packages\\pyls\\language_server.py\", line 65, in wrapped\r","    return func(*args, **kwargs)\r","  File \"c:\\python27\\lib\\site-packages\\pyls\\python_ls.py\", line 111, in m_text_document__did_open\r","    self.lint(textDocument['uri'])\r","  File \"c:\\python27\\lib\\site-packages\\pyls\\python_ls.py\", line 93, in lint\r","    self._hooks.pyls_lint, doc_uri\r","  File \"c:\\python27\\lib\\site-packages\\pyls\\python_ls.py\", line 58, in _hook\r","    return hook(config=self.config, workspace=self.workspace, document=doc, **kwargs)\r","  File \"c:\\python27\\lib\\site-packages\\pluggy.py\", line 745, in __call__\r","    return self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)\r","  File \"c:\\python27\\lib\\site-packages\\pluggy.py\", line 339, in _hookexec\r","    return self._inner_hookexec(hook, methods, kwargs)\r","  File \"c:\\python27\\lib\\site-packages\\pluggy.py\", line 302, in __call__\r","    return outcome.get_result()\r","  File \"c:\\python27\\lib\\site-packages\\pluggy.py\", line 280, in get_result\r","    _reraise(*ex)  # noqa\r","  File \"c:\\python27\\lib\\site-packages\\pluggy.py\", line 265, in __init__\r","    self.result = func()\r","  File \"c:\\python27\\lib\\site-packages\\pluggy.py\", line 300, in <lambda>\r","    outcome = _CallOutcome(lambda: self.oldcall(hook, hook_impls, kwargs))\r","  File \"c:\\python27\\lib\\site-packages\\pluggy.py\", line 334, in <lambda>\r","    _MultiCall(methods, kwargs, hook.spec_opts).execute()\r","  File \"c:\\python27\\lib\\site-packages\\pluggy.py\", line 614, in execute\r","    res = hook_impl.function(*args)\r","  File \"c:\\python27\\lib\\site-packages\\pyls\\plugins\\pycodestyle_lint.py\", line 15, in pyls_lint\r","    config_files = config.find_parents(document.path, CONFIG_FILES)\r","  File \"c:\\python27\\lib\\site-packages\\pyls\\config.py\", line 37, in find_parents\r","    return find_parents(self.root_path, path, names)\r","  File \"c:\\python27\\lib\\site-packages\\pyls\\config.py\", line 70, in find_parents\r","    raise ValueError(\"Path %s not in %s\" % (path, root))\r","ValueError: Path /D:/tmp/python-helloworld/helloworld.py not in file:///D:/tmp/python-helloworld\r",""]]
8/5/2017 3:59:37 PM:[{"response":{"data":{"__data__":"vim-lsp","server_name":"pyls"},"message":"server already started"}}]
8/5/2017 3:59:37 PM:[{"response":{"data":{"__data__":"vim-lsp","init_result":{"id":1,"jsonrpc":"2.0","result":{"capabilities":{"codeLensProvider":{"resolveProvider":false},"executeCommandProvider":{"commands":[]},"hoverProvider":true,"documentFormattingProvider":true,"signatureHelpProvider":{"triggerCharacters":["(",","]},"documentSymbolProvider":true,"definitionProvider":true,"completionProvider":{"resolveProvider":false,"triggerCharacters":["."]},"codeActionProvider":true,"referencesProvider":true,"documentRangeFormattingProvider":true,"textDocumentSync":2}}},"server_name":"pyls"},"message":"lsp server already initialized"}}]
8/5/2017 3:59:37 PM:[{"response":{"data":{"path":"file:///D:/tmp/python-helloworld/helloworld.py","__data__":"vim-lsp","server_name":"pyls"},"message":"already opened"}}]
8/5/2017 3:59:37 PM:[{"response":{"data":{"path":"file:///D:/tmp/python-helloworld/helloworld.py","__data__":"vim-lsp","server_name":"pyls"},"message":"not dirty"}}]
8/5/2017 3:59:37 PM:["--->",1,"pyls",{"method":"textDocument/documentSymbol","on_notification":"---funcref---","params":{"textDocument":{"uri":"file:///D:/tmp/python-helloworld/helloworld.py"}}}]
8/5/2017 3:59:37 PM:["<---",1,"pyls",{"response":{"id":2,"jsonrpc":"2.0","result":[{"location":{"uri":"file:///D:/tmp/python-helloworld/helloworld.py","range":{"end":{"character":13,"line":1},"start":{"character":6,"line":1}}},"name":"MyClass","kind":5},{"location":{"uri":"file:///D:/tmp/python-helloworld/helloworld.py","range":{"end":{"character":12,"line":2},"start":{"character":4,"line":2}}},"name":"variable","kind":13},{"location":{"uri":"file:///D:/tmp/python-helloworld/helloworld.py","range":{"end":{"character":9,"line":7},"start":{"character":0,"line":7}}},"name":"myobjectx","kind":13}]},"request":{"method":"textDocument/documentSymbol","jsonrpc":"2.0","id":2,"params":{"textDocument":{"uri":"file:///D:/tmp/python-helloworld/helloworld.py"}}}}]
8/5/2017 3:59:37 PM:["s:on_text_document_did_open()",2]
8/5/2017 3:59:43 PM:["s:on_text_document_did_close()",2]
8/5/2017 3:59:46 PM:[{"response":{"data":{"__data__":"vim-lsp","server_name":"pyls"},"message":"server already started"}}]
8/5/2017 3:59:46 PM:[{"response":{"data":{"__data__":"vim-lsp","init_result":{"id":1,"jsonrpc":"2.0","result":{"capabilities":{"codeLensProvider":{"resolveProvider":false},"executeCommandProvider":{"commands":[]},"hoverProvider":true,"documentFormattingProvider":true,"signatureHelpProvider":{"triggerCharacters":["(",","]},"documentSymbolProvider":true,"definitionProvider":true,"completionProvider":{"resolveProvider":false,"triggerCharacters":["."]},"codeActionProvider":true,"referencesProvider":true,"documentRangeFormattingProvider":true,"textDocumentSync":2}}},"server_name":"pyls"},"message":"lsp server already initialized"}}]
8/5/2017 3:59:46 PM:[{"response":{"data":{"path":"file:///D:/tmp/python-helloworld/helloworld.py","__data__":"vim-lsp","server_name":"pyls"},"message":"already opened"}}]
8/5/2017 3:59:46 PM:[{"response":{"data":{"path":"file:///D:/tmp/python-helloworld/helloworld.py","__data__":"vim-lsp","server_name":"pyls"},"message":"not dirty"}}]
8/5/2017 3:59:46 PM:["--->",1,"pyls",{"method":"textDocument/definition","on_notification":"---funcref---","params":{"textDocument":{"uri":"file:///D:/tmp/python-helloworld/helloworld.py"},"position":{"character":15,"line":7}}}]
8/5/2017 3:59:46 PM:["<---",1,"pyls",{"response":{"id":3,"jsonrpc":"2.0","result":[{"uri":"file:///D:\\D:\\tmp\\python-helloworld\\helloworld.py","range":{"end":{"character":13,"line":1},"start":{"character":6,"line":1}}}]},"request":{"method":"textDocument/definition","jsonrpc":"2.0","id":3,"params":{"textDocument":{"uri":"file:///D:/tmp/python-helloworld/helloworld.py"},"position":{"character":15,"line":7}}}}]
8/5/2017 3:59:46 PM:["s:on_text_document_did_open()",4]
```

**Comment by tomv564** [comment]
Looks like an issue in `Workspace.get_uri_like`. (also used in textDocument/references)
I'll have a look!

**Comment by gatesn** [comment]
Fixed in #87
